src/data_gradients/managers/abstract_manager.py

```diff
@@ -95,38 +97,88 @@ def execute(self):
             "computer vision datasets. click here: https://hubs.ly/Q01XpHBT0"
         )
 
-        datasets_tqdm = tqdm(
-            zip_longest(self.train_samples_iterator, self.val_samples_iterator, fillvalue=None),
-            desc="Analyzing... ",
-            total=self.n_batches,
-        )
+        # datasets_tqdm = iter(tqdm(
+        #     zip_longest(self.train_samples_iterator, self.val_samples_iterator, fillvalue=None),
+        #     desc="Analyzing... ",
+        #     total=self.n_batches,
+        # ))
+
+        train_samples_iterator = iter(self.train_samples_iterator)
+        val_samples_iterator = iter(self.val_samples_iterator)
 
         self._train_iters_done, self._val_iters_done = 0, 0
         self._stopped_early = False
 
-        for i, (train_sample, val_sample) in enumerate(datasets_tqdm):
+        non_finite_train_samples = 0
+        non_finite_train_samples_example = ""
+        non_finite_val_samples = 0
+        non_finite_val_samples_example = ""
+
+        with tqdm(total=self.n_batches, desc="Analyzing... ") as datasets_tqdm:
+            i = 0
+            while True:
```

Comment:
why not `while i < self.batches_early_stop`

Reply:
The existing logic of the code uses `_stopped_early` flag that we set here to indicate in a report that only part of dataset was processed.
The logic is a bit croocked but I was not feeling confident enough to start refactoring additional piece of logic.